rust/src/tests/mknod.rs

```diff
@@ -188,5 +189,26 @@ fn create_old_new_device(ctx: &mut TestContext) {
     }
 }
 
+// mknod/02.t
+enametoolong_comp_test_case!(mknod(~path, SFlag::S_IFIFO, Mode::empty(), 0));
+
+// mknod/03.t
+enametoolong_path_test_case!(mknod(~path, SFlag::S_IFIFO, Mode::empty(), 0));
+
 // mknod/04.t
 enoent_comp_test_case!(mknod(~path, SFlag::S_IFIFO, Mode::empty(), 0));
+
+mod privileged {
+    use super::*;
+
+    fn mknod_block_wrapper(_: &mut TestContext, path: &Path) -> nix::Result<()> {
+        mknod(path, SFlag::S_IFBLK, Mode::empty(), 0)
+    }
+
+    fn mknod_char_wrapper(_: &mut TestContext, path: &Path) -> nix::Result<()> {
+        mknod(path, SFlag::S_IFCHR, Mode::empty(), 0)
+    }
+
+    // mknod/02.t
+    enametoolong_comp_test_case!(mknod, mknod_block_wrapper, mknod_char_wrapper; root);
```

Comment:
Really?  mknod needs root privileges in this case but not in the whole-path case?

Reply:
Oh no, that's a mistake on my part!